api/documents/controllers.py

```diff
@@ -33,3 +34,17 @@ def delete_document_controller(id: int) -> HTTPStatus:
     document = Document.objects.get(id=id)
     document.delete()
     return HTTPStatus.OK
+
+
+def compare_documents_controller(payload: DocumentIn) -> DocumentOut:
+    embeddings = [doc.embedding for doc in payload]
+    similarities = cosine_similarity(embeddings)
+    result = []
+    for i, doc in enumerate(payload):
+        similar_docs = [
+            {"text": payload[j].text, "similarity": similarities[i][j]}
+            for j in range(len(payload))
+            if i != j
+        ]
+        result.append({"text": doc.text, "similar_documents": similar_docs})
+    return result
```

Comment:
1. This can be optimized, as the similarities matrix is symmetrical, so you don't have to go through both (i, j) and (j, i)
2. I'm pretty sure the result doesn't match DocumentOut schema, did you test this?